### DISCUSSION TITLE
Ability to install to global Python directories

### DISCUSSION THREAD

**Discussion by pgjones** [pdm-project/pdm#2277] [Q&A] [answered]
I've found using PDM in a aws/lambda/python docker image to be difficult as AWS expects the docker CMD to run in the global python install. In general AWS just seems to want to support pip. The poetry solution is to disable [virtualenv creation](https://www.fpgmaas.com/blog/aws-cdk-lambdas-docker) and install everything globally. As far as I can tell this isn't possible with PDM though.

**Comment by frostming** [comment, marked as answer]
`--global --project .` will do the trick.

**Reply by pgjones** [reply]
Ta

**Reply by goyalyashpal** [reply]
@pgjones can you convert this discussion to "Questions and Answers" and then "accept above as answer" ??

<img src="https://github.com/pdm-project/pdm/assets/19423063/dfff4aa7-f2d3-4884-a745-fa55dff12f66" width=250px />

**Comment by ss108** [comment]
I apologize if it's inappropriate/unwelcome to revive this thread, but, from what I can tell, there is no way to simply "turn on" `--global --project .` via configuration, is there? 

Currently, to use PDM to install everything globally, one would must add `--global --project .` to every command individually, which feels sad (and scripting around it, as I am currently doing feels hacky). 

Would you be open to a PR that attempts to add such configuration, or would that be too much for a new contributor, and/or too low priority for you expend the effort reviewing?

**Reply by frostming** [reply]
https://pdm-project.org/latest/usage/config/#passing-constant-arguments-to-every-pdm-invocation

**Reply by ss108** [reply]
oh cool, thanks! Seems to work

**Reply by sh-at-cs** [reply]
Is there a way to set this via `pdm config` or does it have to be placed in `pyproject.toml`?

**Reply by frostming** [reply]
No, applying it globally beats the goal of PDM to isolate your project environments.

**Reply by Yuhaoying** [reply]
在全球范围内应用它会违背
